### PR TITLE
Mongo7 upgrade

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: src/test/java/us/kbase/test/groups/storage/mongo/MongoGroupsStorageStartupTest.java [hunk 1]
```diff
@@ -107,9 +107,10 @@ public void startUpWith2ConfigDocs() throws Exception {
 		
 		final Pattern errorPattern = Pattern.compile(
 				"Failed to create index: Write failed with error code 11000 and error message " +
-				"'(exception: )?E11000 duplicate key error (index|collection): " +
+				"'(exception: )?(.*)E11000 duplicate key error (index|collection): " +
```
Comment (MrCreosote) [inline src/test/java/us/kbase/test/groups/storage/mongo/MongoGroupsStorageStartupTest.java:110]: What is captured by `(.*)`?

File: src/main/java/us/kbase/groups/build/GroupsBuilder.java [hunk 2]
```diff
@@ -106,15 +107,16 @@ public GroupsBuilder(final GroupsConfig cfg, final MongoClient mc)
 	
 	private MongoClient buildMongo(final GroupsConfig c) throws StorageInitException {
 		//TODO ZLATER MONGO handle shards & replica sets
+		final MongoClientSettings.Builder mongoBuilder = MongoClientSettings.builder().applyToClusterSettings(
+				builder -> builder.hosts(Arrays.asList(new ServerAddress(c.getMongoHost()))));
```
Comment (MrCreosote) [inline src/main/java/us/kbase/groups/build/GroupsBuilder.java:111]: You're going to need the retryWrites parameter as well

**Review by MrCreosote — COMMENTED** [review]
Looks pretty good, just needs retryWrites and I have 1 question

File: src/test/java/us/kbase/test/groups/storage/mongo/MongoGroupsStorageStartupTest.java (hunk 1 above)
Comment (Xiangs18) [reply src/test/java/us/kbase/test/groups/storage/mongo/MongoGroupsStorageStartupTest.java:110]: https://github.com/kbase/groups/actions/runs/10517326569/job/29141211739#step:5:310

File: src/main/java/us/kbase/groups/build/GroupsBuilder.java (hunk 2 above)
Comment (Xiangs18) [reply src/main/java/us/kbase/groups/build/GroupsBuilder.java:111]: Adding the retryWrites parameter will be included in the next PR.

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**